### ISSUE TITLE
Auto padding does not take axis title width into account

### ISSUE THREAD

**Issue by kanitw** [vega/vega#1078]
![vega_editor](https://user-images.githubusercontent.com/111269/33142578-decf418c-cf6b-11e7-839e-04decbe0052b.png)

Users can manually set padding as in https://github.com/vega/vega-lite/issues/3207#issuecomment-346428043, but I think we better make the auto padding correct the first time. 

```
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "description": "A simple bar chart with embedded data.",
  "autosize": "pad",
  "padding": 5,
  "height": 200,
  "style": "cell",
  "data": [
    {
      "name": "source_0",
      "values": [
        {"abcdefghijklmnop": "A","b": 28},
        {"abcdefghijklmnop": "B","b": 55}
      ]
    },
    {
      "name": "data_0",
      "source": "source_0",
      "transform": [
        {
          "type": "formula",
          "expr": "toNumber(datum[\"b\"])",
          "as": "b"
        },
        {
          "type": "filter",
          "expr": "datum[\"b\"] !== null && !isNaN(datum[\"b\"])"
        }
      ]
    }
  ],
  "signals": [
    {"name": "x_step","value": 21},
    {
      "name": "width",
      "update": "bandspace(domain('x').length, 0.1, 0.05) * x_step"
    }
  ],
  "marks": [
    {
      "name": "marks",
      "type": "rect",
      "style": ["bar"],
      "from": {"data": "data_0"},
      "encode": {
        "update": {
          "x": {"scale": "x","field": "abcdefghijklmnop"},
          "width": {"scale": "x","band": true},
          "y": {"scale": "y","field": "b"},
          "y2": {"scale": "y","value": 0},
          "fill": {"value": "#4c78a8"}
        }
      }
    }
  ],
  "scales": [
    {
      "name": "x",
      "type": "band",
      "domain": {
        "data": "data_0",
        "field": "abcdefghijklmnop",
        "sort": true
      },
      "range": {"step": {"signal": "x_step"}},
      "paddingInner": 0.1,
      "paddingOuter": 0.05
    },
    {
      "name": "y",
      "type": "linear",
      "domain": {"data": "data_0","field": "b"},
      "range": [{"signal": "height"},0],
      "nice": true,
      "zero": true
    }
  ],
  "axes": [
    {
      "scale": "x",
      "orient": "bottom",
      "labelOverlap": true,
      "title": "abcdefghijklmnop",
      "zindex": 1,
      "encode": {
        "labels": {
          "update": {
            "angle": {"value": 270},
            "align": {"value": "right"},
            "baseline": {"value": "middle"}
          }
        }
      }
    },
    {
      "scale": "y",
      "orient": "left",
      "labelOverlap": true,
      "tickCount": {"signal": "ceil(height/40)"},
      "title": "b",
      "zindex": 1
    },
    {
      "scale": "y",
      "orient": "left",
      "domain": false,
      "grid": true,
      "labels": false,
      "maxExtent": 0,
      "minExtent": 0,
      "tickCount": {"signal": "ceil(height/40)"},
      "ticks": false,
      "zindex": 0,
      "gridScale": "x"
    }
  ],
  "config": {"axisY": {"minExtent": 30}}
}
```

**Comment by jheer** [comment]
Definitely a bug in the ViewLayout transform. Will fix soon.

**Comment by jheer** [comment]
Fixed in vega-view-transforms v1.1.1. Will be included in vega 3.0.8.
